### PR TITLE
Use new Azure SDK

### PR DESCRIPTION
**Pull request by tabman83**
Use the new Azure SDK to load from KV. Addresses #61695 from the backlog.

### CODE REVIEW

File: src/Eshopworld.DevOps/AzureKeyVault/AzureKeyVaultConfigurationProvider.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
+﻿using Azure.Security.KeyVault.Secrets;
+using Microsoft.Extensions.Configuration;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Eshopworld.DevOps.AzureKeyVault
+{
+    public class AzureKeyVaultConfigurationProvider : ConfigurationProvider
```
Comment (mszymeshop) [inline src/Eshopworld.DevOps/AzureKeyVault/AzureKeyVaultConfigurationProvider.cs:9]: Looks like new code, can you explain why we're adding it?

File: src/Eshopworld.DevOps/AzureKeyVault/AzureServiceTokenCredential.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Microsoft.Azure.Services.AppAuthentication;
+
+namespace Eshopworld.DevOps.AzureKeyVault
+{
+    internal class AzureServiceTokenCredential : TokenCredential
```
Comment (mszymeshop) [inline src/Eshopworld.DevOps/AzureKeyVault/AzureServiceTokenCredential.cs:9]: This class was not supposed to be shared (it was a private class originally). For instance expiration time is set as a const.

File: src/Eshopworld.DevOps/Eshopworld.DevOps.csproj [hunk 3]
```diff
@@ -1,13 +1,13 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <Version>5.0.1</Version>
-    <PackageReleaseNotes>Downgrade of Microsoft.Extensions.Configuration.AzureKeyVault</PackageReleaseNotes>
+    <Version>5.1.0</Version>
+    <PackageReleaseNotes>Use new Azure SDK</PackageReleaseNotes>
```
Comment (mszymeshop) [inline src/Eshopworld.DevOps/Eshopworld.DevOps.csproj:5]: Does this PR introduce new SDK and some additional changes?

File: src/Eshopworld.DevOps/AzureKeyVault/AzureKeyVaultConfigurationProvider.cs (hunk 1 above)
Comment (tabman83) [reply src/Eshopworld.DevOps/AzureKeyVault/AzureKeyVaultConfigurationProvider.cs:9]: Former code was based on https://www.nuget.org/packages/Microsoft.Azure.KeyVault/ where is stated that:

> This library has been replaced by the following new Azure SDKs. You can read about the new Azure SDKs at https://aka.ms/azsdkvalueprop.

This commit replaces it with the new SDK as recommended and it solves the tech debt logged in ADO  #61695.

File: src/Eshopworld.DevOps/Eshopworld.DevOps.csproj (hunk 3 above)
Comment (tabman83) [reply src/Eshopworld.DevOps/Eshopworld.DevOps.csproj:5]: Code change is only for new SDK usage in KV loading.

File: src/Eshopworld.DevOps/AzureKeyVault/AzureServiceTokenCredential.cs (hunk 2 above)
Comment (tabman83) [reply src/Eshopworld.DevOps/AzureKeyVault/AzureServiceTokenCredential.cs:9]: Could you please elaborate more? This is new code added with this commit, this class supports the KV reading through the new Azure SDK.

File: src/Eshopworld.DevOps/AzureKeyVault/AzureKeyVaultConfigurationProvider.cs (hunk 1 above)
Comment (mszymeshop) [reply src/Eshopworld.DevOps/AzureKeyVault/AzureKeyVaultConfigurationProvider.cs:9]: Some unit tests maybe?

File: src/Eshopworld.DevOps/AzureKeyVault/AzureServiceTokenCredential.cs [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    {
+        public override async ValueTask<AccessToken> GetTokenAsync(TokenRequestContext requestContext, CancellationToken cancellationToken)
+        {
+            var token = await new AzureServiceTokenProvider().GetAccessTokenAsync("https://vault.azure.net", string.Empty).ConfigureAwait(false);
+            return new AccessToken(token, DateTimeOffset.UtcNow.AddMinutes(5.0));
```
Comment (mszymeshop) [inline src/Eshopworld.DevOps/AzureKeyVault/AzureServiceTokenCredential.cs:14]: Can you check if the 5.0 const here make sense and maybe there is a better and dynamic way to approach this. This is my code so I know I couldn't find much and decided to go with 5 minutes.

Comment (tabman83) [reply src/Eshopworld.DevOps/AzureKeyVault/AzureServiceTokenCredential.cs]: I have changed it to use DefaultAzureCredential which uses a strategy to get access credentials, see below:
https://docs.microsoft.com/en-us/dotnet/api/azure.identity.defaultazurecredential?view=azure-dotnet

File: src/Eshopworld.DevOps/AzureKeyVault/AzureServiceTokenCredential.cs (hunk 2 above)
Comment (tabman83) [reply src/Eshopworld.DevOps/AzureKeyVault/AzureServiceTokenCredential.cs]: I have changed the SecretClient to use DefaultAzureCredential which uses a strategy to get access credentials, see below. Therefore this class is not needed anymore.

https://docs.microsoft.com/en-us/dotnet/api/azure.identity.defaultazurecredential?view=azure-dotnet

**Comment by tabman83** [comment]
I am closing this in favor of https://github.com/eShopWorld/devops/pull/27 which uses a simpler approach.

**Pull request closed without merging**